packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/LinkToCategoriesTests.java

```diff
@@ -0,0 +1,350 @@
+    public void testLinkContentToCategory_usingNonExistingCategoryAndExpect404()
+    {
+        STEP("Try to link content to non-existing category and expect 404");
+        final String nonExistingCategoryId = "non-existing-dummy-id";
+        final RestCategoryLinkBodyModel categoryLink = createCategoryLinkWithId(nonExistingCategoryId);
+        restClient.authenticateUser(user).withCoreAPI().usingNode(file).linkToCategory(categoryLink);
+
+        restClient.assertStatusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Try to link content to category passing empty list as input and expect 400 (Bad Request)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_passingEmptyListAndExpect400()
+    {
+        STEP("Try to call link content API with empty list and expect 400");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(file).linkToCategories(Collections.emptyList());
+
+        restClient.assertStatusCodeIs(BAD_REQUEST);
+    }
+
+    /**
+     * Try to link content to category passing invalid ID in input list and expect 400 (Bad Request)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_passingEmptyIdAndExpect400()
+    {
+        STEP("Try to call link content API with empty category ID and expect 400");
+        final String nonExistingCategoryId = StringUtils.EMPTY;
+        final RestCategoryLinkBodyModel categoryLink = createCategoryLinkWithId(nonExistingCategoryId);
+        restClient.authenticateUser(user).withCoreAPI().usingNode(file).linkToCategory(categoryLink);
+
+        restClient.assertStatusCodeIs(BAD_REQUEST);
+    }
+
+    /**
+     * Link folder node to category and expect 201 (Created)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkFolderToCategory()
+    {
+        STEP("Link folder node to category");
+        final RestCategoryLinkBodyModel categoryLink = createCategoryLinkWithId(category.getId());
+        restClient.authenticateUser(user).withCoreAPI().usingNode(folder).linkToCategory(categoryLink);
+
+        restClient.assertStatusCodeIs(CREATED);
+    }
+
+    /**
+     * Try to link non-content node to category and expect 405 (Method Not Allowed)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_usingTagInsteadOfContentAndExpect405()
+    {
+        STEP("Try to link a tag to category and expect 405");
+        final RestCategoryLinkBodyModel categoryLink = createCategoryLinkWithId(category.getId());
+        final RestTagModel tag = restClient.authenticateUser(user).withCoreAPI().usingNode(file).addTag("someTag");
+        final RepoTestModel tagNode = new RepoTestModel() {};
+        tagNode.setNodeRef(tag.getId());
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingNode(tagNode).linkToCategory(categoryLink);
+
+        restClient.assertStatusCodeIs(METHOD_NOT_ALLOWED);
```

Comment:
`METHOT_NOT_ALLOWED` - to maintain consistency with tags API.

Reply:
This is fine, but is it documented in the api-explorer?

Reply:
I see the point of being consistent with Tags API but I don't think this status code is meant for that. AFAIR, it should be returned when http method is not handled/accepted by the server. 
TBH, I don't think we should repeat mistake we made with tags and we should either return `400 Bad request` or `422 Unprocessable entity`

Reply:
As discussed - I believe we can move on with this code and log a ticket to get this on and (possibly tags) fixed to return a more suitable error code.

Reply:
Ticked created: https://alfresco.atlassian.net/browse/ACS-4362